src/node/desktop/src/main/detect_r.ts

```diff
@@ -37,116 +42,98 @@ function showRNotFoundError(msg: string): void {
  * @returns true if startup should continue, false on fatal error
  */
 export async function prepareEnvironment(): Promise<boolean> {
-  if (process.platform === 'win32') {
-    return await prepareEnvironmentWin32();
-  } else {
-    return await prepareEnvironmentPosix();
-  }
-}
 
-async function prepareEnvironmentPosix(): Promise<boolean> {
-
-  assert((process.platform !== 'win32'));
-
-  // check for which R override
-  let rWhichRPath = new FilePath();
-  const whichROverride = getenv('RSTUDIO_WHICH_R');
-  if (whichROverride) {
-    rWhichRPath = new FilePath(whichROverride);
+  try {
+    return prepareEnvironmentImpl();
```

Comment:
Again, because you're calling an async function (`prepareEnvironmentImpl`) (but **not** using `await` to invoke it), it will never throw an exception! Any exceptions thrown inside it will be converted into a Promise.reject.

If you did use `await` then it could throw an exception. It all makes sense, eventually, but feel like I should make an infographic for it (including for myself).